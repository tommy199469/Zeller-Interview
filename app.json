{
  "name": "QuizApp",
  "displayName": "QuizApp",
  "expo": {
    "name": "QuizApp",
    "slug": "QuizApp",
    "scheme": "quizapp",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/app-icon-all.png",
    "splash": {
      "image": "./assets/images/splash-logo-all.png",
      "resizeMode": "contain",
      "backgroundColor": "#191015"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "jsEngine": "hermes",
    "assetBundlePatterns": [
      "**/*"
    ],
    "android": {
      "icon": "./assets/images/app-icon-android-legacy.png",
      "package": "com.quizapp",
      "adaptiveIcon": {
        "foregroundImage": "./assets/images/app-icon-android-adaptive-foreground.png",
        "backgroundImage": "./assets/images/app-icon-android-adaptive-background.png"
      },
      "splash": {
        "image": "./assets/images/splash-logo-android-universal.png",
        "resizeMode": "contain",
        "backgroundColor": "#191015"
      }
    },
    "ios": {
      "icon": "./assets/images/app-icon-ios.png",
      "supportsTablet": true,
      "bundleIdentifier": "com.quizapp",
      "splash": {
        "image": "./assets/images/splash-logo-ios-mobile.png",
        "tabletImage": "./assets/images/splash-logo-ios-tablet.png",
        "resizeMode": "contain",
        "backgroundColor": "#191015"
      }
    },
    "web": {
      "favicon": "./assets/images/app-icon-web-favicon.png",
      "splash": {
        "image": "./assets/images/splash-logo-web.png",
        "resizeMode": "contain",
        "backgroundColor": "#191015"
      },
      "bundler": "metro"
    },
    "plugins": [
      "expo-localization",
      [
        "expo-build-properties",
        {
          "ios": {
            "newArchEnabled": false
          },
          "android": {
            "newArchEnabled": false
          }
        }
      ],
      "expo-font"
    ],
    "experiments": {
      "tsconfigPaths": true
    }
  },
  "ignite": {
    "version": "9.9.1"
  }
}