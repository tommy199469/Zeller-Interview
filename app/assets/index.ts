const HomeIcon = require("./Icons/home.png")
const CustomerIcon = require("./Icons/customer.png")

export { HomeIcon, CustomerIcon }
