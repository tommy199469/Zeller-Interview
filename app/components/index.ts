export * from "./Screen"
export * from "./DividerLine"
export * from "./RadioGroup"
export * from "./CustomerCard"
export * from "./SearchInput"
export * from "./TabBarLabel"
