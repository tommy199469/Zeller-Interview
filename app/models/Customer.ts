interface ICustomer {
  id: String
  name: String
  email: String
  role: String
}

export { ICustomer }
