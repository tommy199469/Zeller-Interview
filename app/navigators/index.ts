export * from "./AppNavigator"
export * from "./navigationUtilities"
export * from "./NavigationBar"
// export other navigators from here
