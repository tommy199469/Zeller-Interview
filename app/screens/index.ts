export * from "./CustomerScreen"
// export other screens here
