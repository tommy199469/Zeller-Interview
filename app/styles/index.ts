// global style
import { ViewStyle, TextStyle } from "react-native"
import { spacing, colors } from "../theme"
const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.background,
  marginTop: "10%",
}

const $viewContainer: ViewStyle = {
  paddingHorizontal: spacing.lg,
}

const $title: TextStyle = {
  fontWeight: "600",
  fontSize: 24,
}

export { $container, $title, $viewContainer }
