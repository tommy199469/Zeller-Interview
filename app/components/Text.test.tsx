import React from "react"
import { render, fireEvent } from "@testing-library/react-native"
import { CustomerCard, DividerLine, RadioGroup, SearchInput, TabBarIcon, TabBarLabel } from "./"
import { ICustomer } from "../models/Customer"
import { HomeIcon } from "../assets"
import { colors } from "app/theme"
describe("CustomerCard", () => {
  it("renders customer card with name and role", () => {
    const customer: ICustomer = {
      id: "001",
      name: "John Doe",
      role: "Manager",
      email: "test@zeller.com",
    }
    const { getByText } = render(<CustomerCard item={customer} />)
    expect(getByText("John Doe")).toBeDefined()
    expect(getByText("Manager")).toBeDefined()
  })

  it("renders first character of the customer name in icon", () => {
    const customer: ICustomer = {
      id: "001",
      name: "Alice",
      role: "Developer",
      email: "test@zeller.com",
    }
    const { getByText } = render(<CustomerCard item={customer} />)
    expect(getByText("A")).toBeDefined()
  })

  test("renders CustomerCard correctly", () => {
    const customer = { id: "1", name: "John Doe", email: "john@example.com", role: "Manager" }
    const { toJSON } = render(<CustomerCard item={customer} />)
    expect(toJSON()).toMatchSnapshot()
  })
})

describe("DividerLine", () => {
  it("renders divider line", () => {
    const { getByTestId } = render(<DividerLine />)
    expect(getByTestId("divider-line")).toBeDefined()
  })
})

describe("RadioGroup", () => {
  const options = ["Option 1", "Option 2"]
  const selectedValue = "Option 1"
  const onValueChange = jest.fn()

  it("renders radio group options", () => {
    const { getByText, queryByText } = render(
      <RadioGroup options={options} selectedValue={selectedValue} onValueChange={onValueChange} />,
    )
    expect(getByText("Option 1")).toBeDefined()
    expect(getByText("Option 2")).toBeDefined()

    // test for the out of list
    expect(queryByText("Option 3")).toBeNull()
  })

  it("triggers onValueChange when an option is selected", () => {
    const { getByText } = render(
      <RadioGroup options={options} selectedValue={selectedValue} onValueChange={onValueChange} />,
    )
    fireEvent.press(getByText("Option 2"))
    expect(onValueChange).toHaveBeenCalledWith("Option 2")
  })
})

describe("SearchInput", () => {
  const onChangeText = jest.fn()

  it("renders search input with placeholder", () => {
    const { getByPlaceholderText } = render(
      <SearchInput value="" onChangeText={onChangeText} placeholder="Search..." />,
    )

    expect(getByPlaceholderText("Search...")).toBeDefined()
  })

  it("updates the value when text is entered", () => {
    const { getByPlaceholderText } = render(
      <SearchInput value="" onChangeText={onChangeText} placeholder="Search..." />,
    )
    fireEvent.changeText(getByPlaceholderText("Search..."), "test input")
    expect(onChangeText).toHaveBeenCalledWith("test input")
  })
})

describe("TabBarLabel", () => {
  it("renders TabBarLabel with correct text", () => {
    const { getByText } = render(<TabBarLabel text="Home" focused={true} />)
    expect(getByText("Home")).toBeDefined()
  })
})

describe("TabBarIcon", () => {
  const source = HomeIcon

  it("renders TabBarIcon with correct size and tint color", () => {
    const { getByTestId } = render(<TabBarIcon source={source} focused={true} size={30} />)
    const image = getByTestId("icon")
    expect(image.props.style.width).toBe(30)
    expect(image.props.style.height).toBe(30)
    expect(image.props.style.tintColor).toBe(colors.tint) // Assuming `#0171ce` is the tint color
  })
})
