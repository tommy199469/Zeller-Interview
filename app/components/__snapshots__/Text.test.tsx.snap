// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCard renders CustomerCard correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
      },
      {
        "marginTop": 32,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#e8f2fb",
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        },
        {
          "marginRight": 16,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      J
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      John Doe
    </Text>
    <Text
      style={
        [
          {
            "color": "grey",
            "fontSize": 14,
          },
          {
            "marginTop": 12,
          },
        ]
      }
    >
      Manager
    </Text>
  </View>
</View>
`;
